core/src/main/java/com/github/jsonldjava/utils/JsonUtils.java

```diff
@@ -123,6 +123,16 @@ public static Object fromReader(Reader reader) throws IOException {
             throw new JsonParseException("document doesn't start with a valid json element : "
                     + initialToken, jp.getCurrentLocation());
         }
+        
+        JsonToken t ;
+        try { t = jp.nextToken(); }
```

Comment:
The jsonld-java code is directly calling the json parser object which does one level of json object - might be top level, might be nested. So it does not look beyond the closing "}".  I don't see a Jackson call for specifically a top level object and only that but I don't know Jackson/JSON that well.

Majority opinion on the json-ld list is that it is an error to have trailing content.

The JSON spec is quite clear - it's illegal.  The unclear part is (1) a mistake in the JSON-LD spec not being to JSON RFC syntax properly and/or (2) worries about embedded data use cases.  I consider embedding as having a "extract/identify content" step to pull out the JSON.

I think trailing content should be a parse error from jsonld-java.